### PR TITLE
Fix blobber size tests 

### PR DESCRIPTION
**Pull request by Jayashsatolia403**

### CODE REVIEW

**Comment by service-0chain** [comment]
Manual system tests [failure] with the following config
| config  | value |
| ------------- | ------------- |
| tests skipped  | false  |
| system-tests  | sprint-july-1  |
| gosdk  | sprint-july-1  |
| zboxcli  | sprint-july-1  |
| zwalletcli  | sprint-july-1  |
| 0chain  | max_charge  |
| blobber  | sprint-july-1  |
| authorizer  | staging  |
| 0box  | staging  |
| 0dns  | staging  |

**Comment by service-0chain** [comment]
Manual system tests [failure] with the following config
| config  | value |
| ------------- | ------------- |
| tests skipped  | false  |
| system-tests  | fix/blobber_size  |
| gosdk  | NOT OVERRIDDEN  |
| zboxcli  | staging  |
| zwalletcli  | staging  |
| 0chain  | staging  |
| blobber  | fix/cache_allocation  |
| authorizer  | staging  |
| 0box  | staging  |
| 0dns  | staging  |

**Comment by service-0chain** [comment]
Manual system tests [failure] with the following config
| config  | value |
| ------------- | ------------- |
| tests skipped  | false  |
| system-tests  | sprint-july-1  |
| gosdk  | fix/download-filecallback  |
| zboxcli  | sprint-july-1  |
| zwalletcli  | sprint-july-1  |
| 0chain  | staging  |
| blobber  | staging  |
| authorizer  | staging  |
| 0box  | staging  |
| 0dns  | staging  |

**Comment by service-0chain** [comment]
Manual system tests [failure] with the following config
| config  | value |
| ------------- | ------------- |
| tests skipped  | false  |
| system-tests  | sprint-july-1  |
| gosdk  | sprint-july-1  |
| zboxcli  | updategosdk  |
| zwalletcli  | updategosdk  |
| 0chain  | sprint-july-1  |
| blobber  | sprint-july-1  |
| authorizer  | staging  |
| 0box  | staging  |
| 0dns  | staging  |

File: tests/cli_tests/zboxcli_file_upload_test.go [hunk 1]
```diff
@@ -476,11 +476,11 @@ func TestUpload(testSetup *testing.T) {
 			"remotepath": "/",
 			"localpath":  filename,
 		}, false)
-		require.NotNil(t, err, strings.Join(output, "\n"))
-		require.True(t, strings.Contains(strings.Join(output, "\n"), "upload_failed"), strings.Join(output, "\n"))
+		require.Nil(t, err, strings.Join(output, "\n"))
```
Comment (Kishan-Dhakan) [inline tests/cli_tests/zboxcli_file_upload_test.go:479]: I'm assuming the functionality has now changed to `can upload file == 100% size of the allocation`? 

Looks weird, if I create allocation of size 1 GB, 2 data 2 parity, then practically I'm storing 4 GB right?

**Review by Kishan-Dhakan — CHANGES_REQUESTED** [review]

File: tests/cli_tests/zboxcli_file_upload_test.go (hunk 1 above)
Comment (Jayashsatolia403) [reply tests/cli_tests/zboxcli_file_upload_test.go:479]: No you are practically storing 2gb and paying for 2 gb.

Comment (Jayashsatolia403) [reply tests/cli_tests/zboxcli_file_upload_test.go:479]: Per blobber size is size of allocation divided by number of data shards.

Comment (Kishan-Dhakan) [reply tests/cli_tests/zboxcli_file_upload_test.go:479]: In that case, please change these 2 cases ("Upload File too large - parity shards take up allocation space - more than half Size of the Allocation Should not Fail when 1 parity shard" and "Upload File too large - parity shards take up allocation space - more than quarter Size of the Allocation Should not Fail when 3 parity shards") to this:
- Upload file 100% of allocation size should pass
- Upload file more than alloc size should fail

Testing for 50% and 75% doesn't make sense now.

Comment (Jayashsatolia403) [reply tests/cli_tests/zboxcli_file_upload_test.go:479]: `Upload file more than alloc size should fail`

This is already covered. 

`Upload file 100% of allocation size should pass`

I am adding case for this. Thanks!!

**Review by stewartie4 — APPROVED** [review]

**Comment by Kishan-Dhakan** [comment]
Passing run: https://github.com/0chain/system_test/actions/runs/5495805823/jobs/10015712756

**Review by Kishan-Dhakan — APPROVED** [review]

**Pull request merged**